win32ss/drivers/miniport/xboxvmp/xboxvmp.c

```diff
@@ -446,11 +463,25 @@ XboxVmpUnmapVideoMemory(
     PVIDEO_MEMORY VideoMemory,
     PSTATUS_BLOCK StatusBlock)
 {
+    PULONG NV2AFrameBuffer = (PULONG)((ULONG_PTR)DeviceExtension->VirtControlStart + NV2A_CONTROL_FRAMEBUFFER_ADDRESS_OFFSET);
+    PHYSICAL_ADDRESS FrameBuffer;
+
     VideoPortUnmapMemory(
         DeviceExtension,
         VideoMemory->RequestedVirtualAddress,
         NULL);
 
+    FrameBuffer.QuadPart = *NV2AFrameBuffer;
+    FrameBuffer.QuadPart -= DeviceExtension->PhysFrameBufferStart.QuadPart;
+
+    VideoPortReleaseCommonBuffer(
+        DeviceExtension,
+        DeviceExtension->DmaAdapter,
+        DeviceExtension->FrameBufferLength,
+        FrameBuffer,
+        DeviceExtension->VirtFrameBufferStart,
+        FALSE);
+
```

Comment:
You're missing a call to `VideoPortGetDmaAdapter` somewhere (not necessarily here, given the function name).

Reply:
Ok, I changed places a bit. Now `VideoPortGetDmaAdapter` is called in `XboxVmpMapVideoMemory`, and `VideoPortPutDmaAdapter` is called in `XboxVmpUnmapVideoMemory`.